### PR TITLE
feat: add Dockerfile and GitHub workflow to publish image

### PR DESCRIPTION
**Pull request by lentidas**
This PR adds a Dockerfile and a GitHub workflow that allows us to package our Java application as a container.

### CODE REVIEW

File: .github/workflows/builder.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+      - name: "Login to GitHub Container Registry"
+        if: github.event_name == 'push' && github.ref_type == 'tag' # Only login if we push a tag
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: "Set up Docker Buildx"
```
Comment (PedroAS7) [inline .github/workflows/builder.yaml:38]: "Buildx"?

File: .github/workflows/builder.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+      - name: "Build and eventually push container image"
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          file: ./Dockerfile
+          platforms: ${{ matrix.platform }}
+          tags: ${{ steps.metadata_extraction.outputs.tags }}
+          labels: ${{ steps.metadata_extraction.outputs.labels }}
+          push: ${{ github.event_name == 'push' && github.ref_type == 'tag' }} # Only push builded image if we push a tag
```
Comment (PedroAS7) [inline .github/workflows/builder.yaml:49]: "built", not "builded"

**Review by PedroAS7 — CHANGES_REQUESTED** [review]

File: .github/workflows/builder.yaml (hunk 1 above)
Comment (lentidas) [reply .github/workflows/builder.yaml:38]: Buildx is their new build environment (if that's what it's called). It allows us to create multi-architecture images.

File: .github/workflows/builder.yaml (hunk 2 above)
Comment (lentidas) [reply .github/workflows/builder.yaml]:
```suggestion
          push: ${{ github.event_name == 'push' && github.ref_type == 'tag' }} # Only push built image if we push a tag
```

Nice catch!

**Pull request merged**